numba/cuda/compiler.py

```diff
@@ -1133,7 +1070,14 @@ def inspect_llvm(self, signature=None):
 
         '''
         if signature is not None:
-            return self.overloads[signature].inspect_llvm()
+            overload = self.overloads[signature]
+            if isinstance(overload, _Kernel):
```

Comment:
This if/else indicates there is a real distinction between the overloads for kernels and device functions, so combining them may be a problem - unless the `_Kernel` class becomes something more generic, like a `_DeviceFunction` (no judgment here made on how sensible / difficult that may be)